core/arch/x86_64/kernel/thread.c

```diff
@@ -541,19 +541,18 @@ void thread_resume_from_rpc(struct thread_smc_args *args)
 	l->flags &= ~THREAD_CLF_TMP;
 
 	if (threads[n].flags & THREAD_FLAGS_COPY_ARGS_ON_RETURN) {
-		if (threads[n].have_user_map) {
-			struct thread_specific_data *tsd = thread_get_tsd();
-
-			tee_ta_update_session_utime_resume();
+		if (threads[n].have_user_map)
 			core_mmu_set_user_map(&threads[n].user_map);
-		}
 
 		threads[n].flags &= ~THREAD_FLAGS_COPY_ARGS_ON_RETURN;
 		assert(threads[n].ta_idx < MAX_TA_IDX);
 		syscall_init(threads[n].svc_sp, threads[n].svc_handle);
 		thread_rpc_resume(threads[n].stack_va_curr[threads[n].ta_idx],
 				args, l->tmp_stack_va_end);
 	} else if (threads[n].flags & THREAD_FLAGS_EXIT_ON_FOREIGN_INTR) {
+		if (threads[n].have_user_map)
```

Comment:
Can you just call the following code once in thread_resume_from_rpc function?
"if (threads[n].have_user_map)
			core_mmu_set_user_map(&threads[n].user_map);
		syscall_init(threads[n].svc_sp, threads[n].svc_handle);"